src/Datagrid/SimplePager.php

```diff
@@ -66,36 +63,38 @@ public function getNbResults(): int
         return $n;
     }
 
-    public function getResults($hydrationMode = null)
+    public function getResults(?int $hydrationMode = null): array
     {
         if ($this->results) {
             return $this->results;
         }
 
-        $this->results = $this->getQuery()->execute([], $hydrationMode);
-        $this->thresholdCount = \count($this->results);
-        if (\count($this->results) > $this->getMaxPerPage()) {
+        $results = $this->getQuery()->execute([], $hydrationMode);
+
+        // doctrine/phpcr-odm returns ArrayCollection
```

Comment:
Shouldnt we change the phpcr-odm code instead to return an array ?

Reply:
I think this happens this way because the odm has already released a next major, while the orm not. But I am not really sure about what I am saying 😄 . Maybe @greg0ire knows better?

Reply:
Didnt see the comment was about doctrine/phpcr-odm.
Then we should maybe keep this indeed.

Maybe it's better to check for Collection instance instead of ArrayCollection ?

Reply:
This comment was added here: https://github.com/sonata-project/SonataAdminBundle/pull/5144

And the return type is just ArrayCollection: https://github.com/doctrine/phpcr-odm/blob/aabf473e496af9c86d5b2389d098010f322876ad/lib/Doctrine/ODM/PHPCR/DocumentManager.php#L336

Reply:
Maybe it is better to change it to Collection. Done